dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStore.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.VectorData;
+using Npgsql;
+
+namespace Microsoft.SemanticKernel.Connectors.Postgres;
+
+/// <summary>
+/// Represents a vector store implementation using PostgreSQL.
+/// </summary>
+public class PostgresVectorStore : IVectorStore
+{
+    private readonly IPostgresVectorStoreDbClient _postgresClient;
+    private readonly NpgsqlDataSource? _dataSource;
+    private readonly PostgresVectorStoreOptions? _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PostgresVectorStore"/> class.
+    /// </summary>
+    /// <param name="connectionString">Postgres database connection string.</param>
+    /// <param name="options">Optional configuration options for this class</param>
+    public PostgresVectorStore(string connectionString, PostgresVectorStoreOptions? options = default)
+    {
+        NpgsqlDataSourceBuilder dataSourceBuilder = new(connectionString);
+        dataSourceBuilder.UseVector();
+        this._dataSource = dataSourceBuilder.Build();
+        this._options = options ?? new PostgresVectorStoreOptions();
+        this._postgresClient = new PostgresVectorStoreDbClient(this._dataSource, this._options.Schema);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PostgresVectorStore"/> class.
+    /// </summary>
+    /// <param name="dataSource">Postgres data source.</param>
+    /// <param name="options">Optional configuration options for this class</param>
+    public PostgresVectorStore(NpgsqlDataSource dataSource, PostgresVectorStoreOptions? options = default)
+    {
+        this._dataSource = dataSource;
+        this._options = options ?? new PostgresVectorStoreOptions();
+        this._postgresClient = new PostgresVectorStoreDbClient(this._dataSource, this._options.Schema);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PostgresVectorStore"/> class.
+    /// </summary>
+    /// <param name="postgresDbClient">An instance of <see cref="IPostgresDbClient"/>.</param>
+    /// <param name="options">Optional configuration options for this class</param>
+    public PostgresVectorStore(IPostgresVectorStoreDbClient postgresDbClient, PostgresVectorStoreOptions? options = default)
```

Comment:
Is the IPostgresVectorStoreDbClient really an internal implementation detail?  Would it make sense to just expose the NpgsqlDataSource as a parameter option?

Reply:
I agree. I'll make it internal but keep the interface, so that tests and other implementations can pass in custom db clients, but only the Npgsql data source will be exposed to users.